libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp

```diff
@@ -349,10 +363,17 @@ void AC_AttitudeControl_Multi::rate_controller_run()
 
     Vector3f gyro_latest = _ahrs.get_gyro_latest();
 
-    _motors.set_roll(get_rate_roll_pid().update_all(_ang_vel_body.x, gyro_latest.x, _motors.limit.roll) + _actuator_sysid.x);
+    float pd_boost = 1.0f;
+
+    // Boost PD on very rapid throttle changes
+    if (_motors.get_throttle_slew_rate() > _boost_threshold) {
+        pd_boost = _pd_boost;
```

Comment:
Need to constrain pd_boost to min 1.0.  Or at the very least protect against it being set <= 0.

Reply:
This step change in gain propagates straight out to the motors.

Reply:
Not sure I follow?

Reply:
Because its either 1 or the param value when it changes it creates a step change in the PD output that is then directly fed to the motors. Generally were very carful not to put steps into the control system if we can avoid it. The target rates are all slew limited and filtered to avoid steps and as you know we also have lots of filtering on the gyro signal. We also have the slew limit stuff in the PID controller that this bypasses. 

Basically its quite unusual to purposely feed a step into a control system like this.

One way to avoid the step would be something like:
```
    if (_motors.get_throttle_slew_rate() > _boost_threshold) {
        pd_boost = MIN(_motors.get_throttle_slew_rate() / _boost_threshold, _pd_boost);
    }
```

You can probably tell, I'm not really a fan of this approach.... I would at least want to try a hard slew limit output throttle first. This change is sort of "if you move the throttle faster then X then boost gains"  that would be "your not allowed to move the throttle faster than X". If my theory in https://github.com/ArduPilot/ardupilot/pull/21366#issuecomment-1207485338 is correct there would be no difference in performance between the two approaches because its a fundamental motor response time limit.

If you or @lthall have any other theory's to the physical phenomenon that is causing this I would be very interested. I'm not sure that fixing it with this method of gain boosting is the best approach, the argument for it seems to be that this is how betaflight do it, hopefully we can come up with a better solution.